context-monitoring/src/main/java/de/atb/context/persistence/monitoring/MonitoringDataRepository.java

```diff
@@ -137,6 +130,7 @@ protected synchronized void persist(final Model monitoringData, final Applicatio
 
         List<Type> result = new ArrayList<Type>();
         Dataset set = getDataSet(businessCase);
+        set.begin(ReadWrite.WRITE);
         Lock lock = set.getLock();
         lock.enterCriticalSection(true);
```

Comment:
Docs for `enterCriticalSection` state:

> The application must call leaveCriticialSection.

But it looks like `lock.leaveCriticalSection()` might not get called if there is an exception. Should this be handled in a `try-catch-finally` block?

Reply:
According to documentation it seems that Lock is not required when using transactions with TDB; therefore removed.